File: src/models/hardware.ts
export type Host = {
    hostname?: string;
    uptime?: number;
    boot_time?: number;
    procs?: number;
    os?: string;
    platform?: string;
    platform_family?: string;
    platform_version?: string;
    kernel_version?: string;
    kernel_arch?: string;
    virtualization_system?: string;
    virtualization_role?: string;
    host_id?: string;
};

export type CPU = {
    count?: number;
    vendor_id?: string;
    family?: string;
    model?: string;
    stepping?: number;
    physical_id?: string;
    core_id?: string;
    cores_count?: number;
    model_name?: string;
    mhz?: number;
    cache_size?: number;
    flags?: string[];
    microcode?: string;
};

File: core/types/hardware.go
// This package contains the types for the hardware service.
// The structs are almost identical to those of the `gopsutil` library,
// but they are copied here to avoid the dependency and to be able to
// add more fields.

package types

type Host struct {
	Hostname             string `json:"hostname,omitempty"`
	Uptime               uint64 `json:"uptime,omitempty"`
	BootTime             uint64 `json:"boot_time,omitempty"`
	Procs                uint64 `json:"procs,omitempty"`
	OS                   string `json:"os,omitempty"`
	Platform             string `json:"platform,omitempty"`
	PlatformFamily       string `json:"platform_family,omitempty"`
	PlatformVersion      string `json:"platform_version,omitempty"`
	KernelVersion        string `json:"kernel_version,omitempty"`
	KernelArch           string `json:"kernel_arch,omitempty"`
	VirtualizationSystem string `json:"virtualization_system,omitempty"`
	VirtualizationRole   string `json:"virtualization_role,omitempty"`
	HostID               string `json:"host_id,omitempty"`
}

type CPU struct {
	Count      int32    `json:"count,omitempty"`
	VendorID   string   `json:"vendor_id,omitempty"`
	Family     string   `json:"family,omitempty"`
	Model      string   `json:"model,omitempty"`
	Stepping   int32    `json:"stepping,omitempty"`
	PhysicalID string   `json:"physical_id,omitempty"`
	CoreID     string   `json:"core_id,omitempty"`
	CoresCount int32    `json:"cores_count,omitempty"`
	ModelName  string   `json:"model_name,omitempty"`
	Mhz        float64  `json:"mhz,omitempty"`
	CacheSize  int32    `json:"cache_size,omitempty"`
	Flags      []string `json:"flags,omitempty"`
	Microcode  string   `json:"microcode,omitempty"`
}
